src/components/SuggestionsList/SuggestionsList.md

```diff
@@ -0,0 +1,120 @@
+### Examples
+
+With no results:
+
+```js { "props": { "data-example": "No Results" } }
+<div style={{ minHeight: '52px' }}>
+  <span id="suggested-list-1" />
+  <SuggestionsList
+    target="#suggested-list-1"
+    noResultsText="No results found"
+  />
+</div>
+```
+
+When loading:
+
+```js { "props": { "data-example": "When Loading" } }
+<div style={{ minHeight: '52px' }}>
+  <span id="suggested-list-2" />
+  <SuggestionsList
+    target="#suggested-list-2"
+    isLoading={true}
+    loadingText={'Searching for \'Cris\''}
+  />
+</div>
+```
+
+With results:
+
+```js { "props": { "data-example": "When Results" } }
+<div style={{ minHeight: '244px' }}>
+  <span id="suggested-list-3" />
+  <SuggestionsList
+    target="#suggested-list-3"
+    searchText="Cris"
+    loadingText={'Searching for \'Cris\''}
+    onItemSelected={() => {}}
+    groupedItems={[
+      {
+        title: 'Suggested Users',
+        items: [{
+          id: 1,
+          imageUrl: '',
+          name: 'Cristiana Cline',
+          description: 'Data Analyst'
+        }, {
+          id: 2,
+          imageUrl: '',
+          name: 'Cristian Sayers',
+          description: 'Front-End Developer'
+        }, {
+          id: 3,
+          imageUrl: '',
+          name: 'Johanna Cristian Lorenz',
+          description: 'Information Architecture'
+        }, {
+          id: 4,
+          imageUrl: '',
+          name: 'Lidia Holloway Cristina',
+          description: 'Senior Designer'
+        }, {
+          id: 5,
+          imageUrl: '',
+          name: 'Nestor Cristian Wilke',
+          description: 'Product Manager'
+        }]
+      }
+    ]}
+    selectedId={2}
+  />
+</div>
+```
+
+With results, when loading:
+
+```js { "props": { "data-example": "When Results" } }
+<div style={{ minHeight: '297px' }}>
+  <span id="suggested-list-4" />
+  <SuggestionsList
+    target="#suggested-list-4"
```

Comment:
Using string refs will get removed at any point: https://reactjs.org/docs/refs-and-the-dom.html#legacy-api-string-refs

Please don't use them even in examples.

Reply:
mm this isnt a ref like the react property ref - this is target which goes to Callout target

Reply:
You sure it's not using a React `ref` under the hood?

Reply:
idk - what would you change this example to? Also - any ideas how to fix the stupid screwy Callout thing, I dont know a good solution.

Reply:
I'd check Fabric's examples, but maybe it'll look like this?

```
<span ref={(target) => { this.target = target; }} />
<SuggestionsList target={this.target} />
```

Reply:
I'll checkout your branch, see what's the screwy behavior.

Reply:
`doNotLayer` causes the Callout to scroll with the page, as it doesn't absolutely position the thing:

```js
      <Callout
        target={this.props.target}
        className={classNames.join(' ')}
        isBeakVisible={false}
        doNotLayer={true}
      >
        {results}
        {status}
      </Callout>
```

Reply:
Is that what you meant?

Reply:
yeah - just need to workout how to give it a non-string ref for the demo then

Reply:
(your example doesnt work cos the ref cb wont resolve in time to pass it to the sibling)

Reply:
You are right. We can keep it as string until React removes it, and then we fix it?

Reply:
i added a wrapper with a trigger button - can you take a look at CalloutTrigger? Having them render by default was causing jump around (cos of the auto repositioning magic of Callout). It's better to have them toggle on and off cos that forces rerender and repositioning
<img width="1008" alt="screen shot 2017-12-14 at 11 47 33 am" src="https://user-images.githubusercontent.com/692240/34011426-d21acc62-e0c4-11e7-82b8-c90cb062cdec.png">
![Uploading Screen Shot 2017-12-14 at 11.47.27 AM.png…]()